SERVER/api/v1/controllers/adCtrl.js

```diff
@@ -47,51 +48,67 @@ const Ad = {
         }
 
-    },
+    }
 
     // User get all unsold cars. 
-    async getAllCars(req, res, next) {
+     static async getAllCars(req, res, next) {
+     
+        try {
 
-        const range = {
-            min: 0,
-            max: 500000000000
-        }
-        const maxCarPrice = "$" + range.max.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ",");
-        const minCarPrice = "$" + range.min.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ",");
+            let  {rows} = await adModel.findPrice(); 
+            const thePrice = rows; 
+            // console.log('The Price:',thePrice);
+
+            if (req.query.status === 'available' && req.query.min_price >=thePrice &&  req.query.max_price <= thePrice){
+
+                // Available cars within a certain range. 
+                const finishedFilter= [req.query.status , req.query.min_price,  req.query.max_price];
+
+                // console.log('The Range:', inSomeRange); 
+                const {inRange} = await adModel.priceRange(finishedFilter);   
+
+                console.log('The rows:', inRange);
+
+                if(!theAVailable[0]) {
+                    return res.status(404).json({
+                        status: 404, 
+                        error: 'No cars in that price range!'
+                    })
+                }
+                return res.status(200).json({
+                    staus: 200,
+                    message: 'Cars within that price range',
+                    data: inRange
+                }); 
+            };
+
+            if (req.query.status === "available"){
 
-        try {
-            if (req.query.status === "available") {
                 // cars with just "available" status.
-                const available = ads.filter(ad => ad.status === "available");
+                const available_cars = req.query.status;
+               const {rows} = await adModel.availableCars(available_cars); 
+
                 return res.status(200).json({
                     status: 200,
                     message: 'Here are all available cars',
-                    data: available
+                    data: rows
 
                 });
             };
 
-            if (req.query.status === "avilable" && "$" + req.query.min_price.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ",") >= minCarPrice && "$" + req.query.max_price.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ",")) {
-                // Available cars within a certain range.    
-                const priceRange = ads.filter(ad => ad.status === "available" && ad.price >= minCarPrice && ad.price <= maxCarPrice);
-                return res.status(200).json({
-                    staus: 200,
-                    message: 'Cars within that price range',
-                    data: priceRange
-                })
-            };
-
-            if (req.user.is_admin === true) {
+            if(req.user.is_admin === true) {
+                
+                const {rows} = await adModel.allCars(); 
```

Comment:
`rows` is not good a variable name to use, consider changing it to a better meaningful name